PluginCore/PluginCore/Controls/ScrollBarEx.cs

```diff
@@ -18,15 +18,95 @@ namespace PluginCore.Controls
     [DefaultProperty("Value")]
     public class ScrollBarEx : Control
     {
+        #region scroll bar settings
+
+        public static bool UseCustom
+        {
+            get { return PluginBase.MainForm.GetThemeValue("ScrollBar.UseCustom", "false").ToLower() == "true"; }
+        }
+
+        #endregion
+
         #region drawing
 
-        private Color curPosColor = Color.DarkBlue;
-        private Color borderColor = SystemColors.ActiveBorder;
-        private Color borderColorDisabled = SystemColors.Control;
-        private Color foreColor = SystemColors.ControlDarkDark;
-        private Color foreColorActive = SystemColors.Highlight;
-        private Color backColor = SystemColors.ActiveBorder;
-        private Color backColorDisabled = SystemColors.ControlLight;
+        private Color curPosColor/* = Color.DarkBlue*/;
+        private Color foreColor/* = SystemColors.ControlDarkDark*/;
+        private Color foreColorHot;
+        private Color foreColorPressed/* = SystemColors.Highlight*/;
+        private Color arrowColor;
+        private Color arrowColorHot;
+        private Color arrowColorPressed;
+        private Color backColor/* = SystemColors.ActiveBorder*/;
+        private Color backColorDisabled/* = SystemColors.ControlLight*/;
+        private Color borderColor/* = SystemColors.ActiveBorder*/;
+        private Color borderColorDisabled/* = SystemColors.Control*/;
+        private Boolean colorsInvalidated = false;
+
+        /// <summary>
+        /// Resets the component colors to default values.
+        /// </summary>
+        private void InitializeColors()
+        {
+            if (curPosColor.IsEmpty)         curPosColor         = Color.FromArgb(  0,   0, 205); //#0000CD
+            if (foreColor.IsEmpty)           foreColor           = Color.FromArgb(208, 209, 215); //#D0D1D7
+            if (!colorsInvalidated)          foreColorHot        = Color.FromArgb(136, 136, 136); //#888888
+            if (foreColorPressed.IsEmpty)    foreColorPressed    = Color.FromArgb(106, 106, 106); //#6A6A6A
+            if (!colorsInvalidated)          arrowColor          = Color.FromArgb(134, 137, 153); //#868999
+            if (!colorsInvalidated)          arrowColorHot       = Color.FromArgb( 28, 151, 234); //#1C97EA
+            if (!colorsInvalidated)          arrowColorPressed   = Color.FromArgb(  0, 122, 204); //#007ACC
+            if (backColor.IsEmpty)           backColor           = Color.FromArgb(232, 232, 236); //#E8E8EC
+            if (backColorDisabled.IsEmpty)   backColorDisabled   = Color.FromArgb(192, 192, 192); //#C0C0C0
+            if (borderColor.IsEmpty)         borderColor         = Color.FromArgb(232, 232, 236); //#E8E8EC
+            if (borderColorDisabled.IsEmpty) borderColorDisabled = Color.FromArgb(136, 136, 136); //#888888
+            colorsInvalidated = false;
```

Comment:
`colosInvalidated = false` causes the fallback colors not being assigned.
I remember changing the structure of `ValidateColors` after testing the themes :(
I'll remove this line.